clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -240,40 +240,47 @@ protected boolean flush(ByteBuffer buf) throws IOException {
     */
     @Override
     public void handshake() throws IOException {
-        int read = 0;
+        // Reset state to support renegotiation. This can be removed if renegotiation support is removed.
+        if (state == State.READY)
+            state = State.HANDSHAKE;
+
         // Read any available bytes before attempting any writes to ensure that handshake failures
         // reported by the peer are processed even if writes fail (since peer closes connection
         // if handshake fails)
+        int read = 0;
         if (key.isReadable())
             read = readFromSocketChannel();
```

Comment:
Could this throw an exception? Should we be doing `maybeThrowSslAuthenticationException` if it does?